packages/plugin-core/src/test/suite-integ/ReferenceHoverProvider.test.ts

```diff
@@ -0,0 +1,1252 @@
+import { NoteProps, VaultUtils } from "@dendronhq/common-all";
+import {
+  AssertUtils,
+  FileTestUtils,
+  NoteTestUtilsV4,
+} from "@dendronhq/common-test-utils";
+import { ENGINE_HOOKS_MULTI } from "@dendronhq/engine-test-utils";
+import fs from "fs-extra";
+import { before, describe } from "mocha";
+import path from "path";
+import * as vscode from "vscode";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import ReferenceHoverProvider from "../../features/ReferenceHoverProvider";
+import { VSCodeUtils } from "../../vsCodeUtils";
+import { getDWorkspace } from "../../workspace";
+import { WSUtils } from "../../WSUtils";
+import { expect, LocationTestUtils } from "../testUtilsv2";
+import {
+  describeMultiWS,
+  describeSingleWS,
+  runLegacyMultiWorkspaceTest,
+  setupBeforeAfter,
+} from "../testUtilsV3";
+
+async function provide({
+  editor,
+  pos,
+}: {
+  editor: vscode.TextEditor;
+  pos: vscode.Position;
+}) {
+  const doc = editor?.document as vscode.TextDocument;
+  const referenceProvider = new ReferenceHoverProvider();
+  return referenceProvider.provideHover(doc, pos);
+}
+
+async function provideForNote(editor: vscode.TextEditor) {
+  return provide({ editor, pos: new vscode.Position(7, 4) });
+}
+
+async function provideForNonNote(editor: vscode.TextEditor) {
+  return provide({ editor, pos: new vscode.Position(0, 2) });
+}
+
+suite("GIVEN ReferenceHoverProvider", function () {
+  const ctx = setupBeforeAfter(this);
+
+  describe("AND GIVEN non-dendron file", () => {
+    describeMultiWS(
+      "AND WHEN try to find references",
```

Comment:
nitpick: I think the messages here could be clearer, like "WHEN used in a non-dendron file", "AND the file is empty", "THEN doesn't provide for this file", so that it reads like a sentence when put together.